test/common/access_log/access_log_formatter_speed_test.cc

```diff
@@ -61,37 +61,38 @@ BENCHMARK(BM_TypedJsonAccessLogFormatter);
 
 } // namespace Envoy
 
-// Boilerplate main(), which discovers benchmarks in the same file and runs them.
-int main(int argc, char** argv) {
-  static const char* LogFormat =
-      "%DOWNSTREAM_REMOTE_ADDRESS_WITHOUT_PORT% %START_TIME(%Y/%m/%dT%H:%M:%S%z %s)% "
-      "%REQ(:METHOD)% "
-      "%REQ(X-FORWARDED-PROTO)%://%REQ(:AUTHORITY)%%REQ(X-ENVOY-ORIGINAL-PATH?:PATH)% %PROTOCOL% "
-      "s%RESPONSE_CODE% %BYTES_SENT% %DURATION% %REQ(REFERER)% \"%REQ(USER-AGENT)%\" - - -\n";
-
-  formatter = std::make_unique<Envoy::AccessLog::FormatterImpl>(LogFormat);
-
-  std::unordered_map<std::string, std::string> JsonLogFormat = {
-      {"remote_address", "%DOWNSTREAM_REMOTE_ADDRESS_WITHOUT_PORT%"},
-      {"start_time", "%START_TIME(%Y/%m/%dT%H:%M:%S%z %s)%"},
-      {"method", "%REQ(:METHOD)%"},
-      {"url", "%REQ(X-FORWARDED-PROTO)%://%REQ(:AUTHORITY)%%REQ(X-ENVOY-ORIGINAL-PATH?:PATH)%"},
-      {"protocol", "%PROTOCOL%"},
-      {"respoinse_code", "%RESPONSE_CODE%"},
-      {"bytes_sent", "%BYTES_SENT%"},
-      {"duration", "%DURATION%"},
-      {"referer", "%REQ(REFERER)%"},
-      {"user-agent", "%REQ(USER-AGENT)%"}};
-
-  json_formatter = std::make_unique<Envoy::AccessLog::JsonFormatterImpl>(JsonLogFormat, false);
-  typed_json_formatter = std::make_unique<Envoy::AccessLog::JsonFormatterImpl>(JsonLogFormat, true);
-
-  stream_info = std::make_unique<Envoy::TestStreamInfo>();
-  stream_info->setDownstreamRemoteAddress(
-      std::make_shared<Envoy::Network::Address::Ipv4Instance>("203.0.113.1"));
-  benchmark::Initialize(&argc, argv);
-  if (benchmark::ReportUnrecognizedArguments(argc, argv)) {
-    return 1;
+// Hack to initialize the global variables used by the benchmarks above.
```

Comment:
FWIW, this extant hack is pretty terrible, I hope we can just get rid of it and not rely on globals.

Reply:
I don't know how to get rid of it.  Do you have any suggestions?

Reply:
Fixed.